backend/apis/leads/add-leads.api.js

```diff
@@ -0,0 +1,112 @@
+const mysql = require("mysql");
+const { app, databaseError, pool, invalidRequest } = require("../../server");
+const {
+  checkValid,
+  nonEmptyString,
+  validArray,
+  validBoolean,
+  validDate,
+  validInteger
+} = require("../utils/validation-utils");
+
+app.post("/api/leads", (req, res) => {
+  const leads = req.body.data;
+  const user = req.session.passport.user;
+
+  for (let i = 0; i < leads.length; i++) {
```

Comment:
this will throw an error when the request does not have a body (`Cannot read property length of undefined`). Also when the request body is not an array.

To fix, add this before your for loop:

```js
if (!Array.isArray(leads)) {
  return invalidRequest(res, [`POST /api/leads must be called with an array of leads in the request body`])
}
```